test/conformance/configuration_test.go

```diff
@@ -0,0 +1,139 @@
+// +build e2e
+
+/*
+Copyright 2018 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package conformance
+
+import (
+	"github.com/knative/pkg/test/logging"
+	"github.com/knative/serving/pkg/apis/serving/v1alpha1"
+	"github.com/knative/serving/test"
+	"k8s.io/apimachinery/pkg/apis/meta/v1"
+	"reflect"
+	"testing"
+)
+
+func TestUpdateConfigurationMetadata(t *testing.T) {
+	clients := setup(t)
+
+	logger := logging.GetContextLogger("TestUpdateConfigurationMetadata")
+
+	var names test.ResourceNames
+	names.Service = test.AppendRandomString("pizzaplanet-service", logger)
+	names.Config = names.Service
+
+	defer tearDown(clients, names)
+	test.CleanupOnInterrupt(func() { tearDown(clients, names) }, logger)
+
+	logger.Infof("Creating new configuration %s", names.Config)
+	err := test.CreateConfiguration(logger, clients, names, test.ImagePath(image1), &test.Options{})
+	if err != nil {
+		t.Fatalf("Failed to create configuration %s", names.Config)
+	}
+
+	cfg := reloadConfiguration(names.Config, clients, t)
+
+	logger.Info("The Configuration will be updated with the name of the Revision once it is created")
+	names.Revision, err = waitForConfigurationLatestCreatedRevision(clients, names)
+	if err != nil {
+		t.Fatalf("Configuration %s was not updated with the new revision: %v", names.Config, err)
+	}
+
+	cfg = reloadConfiguration(names.Config, clients, t)
+
+	logger.Infof("Updating labels of Configuration %s", names.Config)
+	cfg.Labels =  map[string]string{
+		"labelX": "abc",
+		"labelY": "def",
+	}
+	cfg, err = clients.ServingClient.Configs.Update(cfg)
+	if err != nil {
+		t.Fatalf("Failed to update labels for Configuration %s: %v", names.Config, err)
+	}
+
+	if err = waitForConfigurationLabelsUpdate(clients, names, cfg.Labels); err != nil {
```

Comment:
I'm wondering if  we're waiting for the right condition here. Essentially you're waiting for the configuration to actually receive the update on labels (which I'm not sure isn't happening instantly anyway, see below).

Would we need to wait for a certain amount of time and observe the `LatestRevision` field not change?

Reply:
I am a bit confused. I want to verify that a new revision is **not** created. If the update happens asynchronously, there can be an indeterminate amount of time for the update to actually happen. In practice I realize that the updates typically happen very fast.

Doesn't `waitForConfigurationLabelsUpdate` poll until the label updates are finished?

Reply:
Ah, my mistake.  I thought this was verifying that we DO create a new revision when we change: `cfg.spec.revisionTemplate.metadata.{labels,annotations}`

Reply:
I have another [PR](https://github.com/knative/serving/pull/2253) for #2253 that does verify a new revision is created when labels and annotations are updated.